src/System.Net.Http/src/System/Net/Http/Managed/DecompressionHandler.cs

```diff
@@ -47,30 +47,22 @@ public DecompressionHandler(DecompressionMethods decompressionMethods, HttpMessa
 
             HttpResponseMessage response = await _innerHandler.SendAsync(request, cancellationToken).ConfigureAwait(false);
 
-            while (true)
+            ICollection<string> contentEncodings = response.Content.Headers.ContentEncoding;
+            if (contentEncodings.Count > 0)
             {
-                // Get first encoding
-                using (IEnumerator<string> e = response.Content.Headers.ContentEncoding.GetEnumerator())
+                string last = null;
+                foreach (string encoding in contentEncodings)
                 {
-                    if (!e.MoveNext())
-                    {
-                        break;
-                    }
+                    last = encoding;
```

Comment:
https://github.com/stephentoub/corefx/blob/85f72ea9714440c1f4055672025b9c5da6fc2c11/src/System.Net.Http/src/System/Net/Http/Managed/DecompressionHandler.cs#L103 is still removing the first rather than the last.

Reply:
Grrr, yes it is, will fix.